trieste/bayesian_optimizer.py

```diff
@@ -99,14 +151,16 @@ class OptimizationResult(Generic[StateType]):
     exception.
     """
 
-    history: list[Record[StateType]]
+    history: list[Record[StateType] | FrozenRecord[StateType]]
     r"""
     The history of the :class:`Record`\ s from each step of the optimization process. These
-    :class:`Record`\ s are created at the *start* of each loop, and as such will never include the
-    :attr:`final_result`.
+    :class:`Record`\ s are created at the *start* of each loop, and as such will never
+    include the :attr:`final_result`. The records may be either in memory or on disk.
```

Comment:
Im missing something here. How do you specify if will be saved to disk? I guess we dont always want this to happen? Or maybe we do?

Reply:
You specify it in the track_path argument to optimize (as it's BayesianOptimizer that actually generates these). The default is still to track in memory.

Reply:
I've updated the tutorials to explain this.